pkg/asset/installconfig/azure/validation.go

```diff
@@ -137,6 +139,37 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 		}
 	}
 
+	// Convenience check so the tests don't have to define LocationInfo if it's not used
+	if typeMeta.LocationInfo != nil && ultraSSDEnabled {
+		for _, locationInfo := range *typeMeta.LocationInfo {
+			// If Availability Zones are not supported (e.g StackCloud)
+			if locationInfo.ZoneDetails == nil {
+				continue
+			}
```

Comment:
This case is already handled by the preceding conditional two lines earlier.

Reply:
In my tests against the API for the Azure Stack case the `LocationInfo` struct was defined but the `ZoneDetails` field was `nil`. So I need to check that before dereferencing the pointer in the `for` loop that follows. I even added that as a test case https://github.com/openshift/installer/pull/5854/files#diff-310a64406bdb016820cd032ef2ca5cd092ba942f2b7ccbedeec0abc254fbef74R670-R676